### PR TITLE
avoid coercion to block for new plist file content, fix adding a new dictionary 

### PR DESCRIPTION
**Pull request by jazaval**
Signed-off-by: Jacob Zaval <jazava@microsoft.com>

## Description
When creating a new plist, the `plist` resource passes an empty hash with the `to_plist` method (`{}.to_plist`) to the `content` property of `file` resource. However, the `file` resource interprets this first as a block, as demonstrated in #12678. This PR assigns the empty plist to a variable first, so it can be properly evaluated before being passed to `file`.

Also, during the addition of a functional spec test with a `Hash` for `value`, I discovered a bug with the "add" case for a new dictionary and updated accordingly (similar solution to the `Hash` "set" case).

## Related Issue
#12678 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/resource/plist.rb [hunk 1]
```diff
@@ -85,7 +85,8 @@ class PlistResource < Chef::Resource # we name this PlistResource to avoid confu
         converge_if_changed :path do
           converge_by "create new plist: '#{new_resource.path}'" do
             file new_resource.path do
-              content {}.to_plist
```
Comment (lamont-granquist) [inline lib/chef/resource/plist.rb:88]: I think `content({}.to_plist)` should work here to avoid coercing to a block?

Comment (jazaval) [reply lib/chef/resource/plist.rb:88]: Can confirm that works as well.

File: lib/chef/resource/plist.rb [hunk 2]
```diff
@@ -189,7 +189,12 @@ def plistbuddy_command(subcommand, entry, path, value = nil)
         sep = " "
         arg = case subcommand.to_s
               when "add"
-                type_to_commandline_string(value)
+                if value.is_a?(Hash)
+                  sep = ":"
+                  value.map { |k, v| "#{k} #{type_to_commandline_string(v)}" }.join(sep)
```
Comment (lamont-granquist) [inline lib/chef/resource/plist.rb:194]: I don't think you need the `.join(sep)` here?  It isn't in the other branch, and it looks like it falls through and will hit the `.join(sep).strip` after the case statement.

Comment (jazaval) [reply lib/chef/resource/plist.rb]: Yep, you're right. Missed that when porting this from one of our other libraries.

**Comment by jazaval** [comment]
@lamont-granquist test failures seem unrelated to my change - is this okay to merge?

**Review by lamont-granquist — APPROVED** [review]

**Pull request merged**